<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<arg name="veh" default="SQ"/>
	<arg name="num" default="01"/> <!-- use this for snap_sim (if quad is SQ01s then num is 01) -->

	<arg name="x" default="0.0"/>
	<arg name="y" default="0.0"/>
	<arg name="z" default="0.8" />
	<arg name="yaw" default="0.0" />

	<arg name="goal_topic" default="goal"/>

	<!-- Start rmader commands -->
	<node ns="$(arg veh)$(arg num)s" name="rmader_commands" pkg="rmader" type="rmader_commands.py" output="screen"/>

	<!-- Start Perfect tracker or snap_sim -->
	<include file="$(find rmader)/launch/perfect_tracker_and_sim.launch">
  		<arg name="quad" value="$(arg veh)$(arg num)s"/>
  		<arg name="gazebo" value="false"/>
        <arg name="x" value="$(arg x)"/>
        <arg name="y" value="$(arg y)"/>
        <arg name="z" value="$(arg z)"/>
        <arg name="yaw" default="$(arg yaw)"/>
        <!-- <arg name="perfect_tracker" default="false"/> -->
 	 </include>

 	<!-- rmader -->
	<node ns="$(arg veh)$(arg num)s" name="rmader" pkg="rmader" type="rmader_node" output="screen" clear_params="false" >
		<rosparam file="$(find rmader)/param/rmader.yaml" command="load"/>
	        <!-- Subscriptions -->
			<remap from="~state" to="state"/>
			<remap from="~who_plans" to="who_plans"/>
			<remap from="~term_goal" to="term_goal" />

			<!-- Publications -->
			<remap from="~goal" to="$(arg goal_topic)" />
			<remap from="~setpoint" to="setpoint" />
	</node>
</launch>
	